src/utils/defaultFilterOptions.js

```diff
@@ -20,24 +20,35 @@ function filterOptions (options, filterValue, excludeOptions, props) {
 		if (excludeOptions && excludeOptions.indexOf(option[props.valueKey]) > -1) return false;
 		if (props.filterOption) return props.filterOption.call(this, option, filterValue);
 		if (!filterValue) return true;
-		var valueTest = String(option[props.valueKey]);
-		var labelTest = String(option[props.labelKey]);
+
+		var value = option[props.valueKey];
+		var label = option[props.labelKey];
+		var hasValue = typeof (value) !== 'undefined' && value !== null && value !== '';
+		var hasLabel = typeof (label) !== 'undefined' && label !== null && label !== '';
+
+		if (!hasValue && !hasLabel) {
+			return false;
+		}
+
+		var valueTest = hasValue ? String(value) : null;
+		var labelTest = hasLabel ? String(label) : null;
```

Comment:
Why not include this in the hasX checks?

Reply:
It's because there is situation when only one of label or value is invalid and still need to search in valid field.